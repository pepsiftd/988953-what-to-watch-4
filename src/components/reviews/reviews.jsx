import React from 'react';
import PropTypes from 'prop-types';

import {Review} from '@/components/review/review';

const Reviews = ({reviews}) => {
  const columnHeight = Math.ceil(reviews.length / 2);
  return (
    <div className="movie-card__reviews movie-card__row">
      <div className="movie-card__reviews-col">
        {reviews.slice(0, columnHeight).map((review) => (
          <Review key={review.id} review={review} />
        ))}
      </div>
      <div className="movie-card__reviews-col">
        {reviews.slice(columnHeight).map((review) => (
          <Review key={review.id} review={review} />
        ))}
      </div>
    </div>
  );
};

Reviews.propTypes = {
  reviews: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    user: PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
    }).isRequired,
    rating: PropTypes.number.isRequired,
    comment: PropTypes.string.isRequired,
    date: PropTypes.instanceOf(Date),
  })).isRequired,
};

export {Reviews};
