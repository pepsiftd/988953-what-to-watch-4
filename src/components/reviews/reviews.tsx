import * as React from 'react';
import {ReviewData} from '@/types';

import {Review} from '@/components/review/review';

const COLUMNS_AMOUNT = 2;

interface Props {
  reviews: ReviewData[];
}

const Reviews: React.FunctionComponent<Props> = ({reviews}) => {
  const columnHeight = Math.ceil(reviews.length / COLUMNS_AMOUNT);
  return (
    <div className="movie-card__reviews movie-card__row">
      <div className="movie-card__reviews-col">
        {reviews.slice(0, columnHeight).map((review) => (
          <Review key={review.id} review={review} />
        ))}
      </div>
      <div className="movie-card__reviews-col">
        {reviews.slice(columnHeight).map((review) => (
          <Review key={review.id} review={review} />
        ))}
      </div>
    </div>
  );
};

export {Reviews};
