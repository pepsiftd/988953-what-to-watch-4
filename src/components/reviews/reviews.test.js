import React from 'react';
import renderer from 'react-test-renderer';

import {Reviews} from './reviews';

const reviews = [
  {
    id: 1,
    user: {
      id: 4,
      name: `Courtney Cox`,
    },
    rating: 5.1,
    comment: `I don't know what to say but I can't help posting`,
    date: new Date(`August 3, 2020`),
  },
  {
    id: 2,
    user: {
      id: 2,
      name: `Mockingbird`,
    },
    rating: 8.3,
    comment: `This is not a movie of a simple kind`,
    date: new Date(`July 15, 2020`),
  },
  {
    id: 3,
    user: {
      id: 154,
      name: `Pelagia`,
    },
    rating: 1.5,
    comment: `Super mega exquisitely boring`,
    date: new Date(`June 1, 2020`),
  }
];

it(`Reviews renders correctly`, () => {
  const tree = renderer.create(<Reviews reviews={reviews} />);

  expect(tree).toMatchSnapshot();
});
