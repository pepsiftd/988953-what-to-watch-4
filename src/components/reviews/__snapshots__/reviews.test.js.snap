// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews renders correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          I don't know what to say but I can't help posting
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Courtney Cox
          </cite>
          <time
            className="review__date"
            dateTime="2020-08-02T21:00:00.000Z"
          >
            August 3, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        5,1
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          This is not a movie of a simple kind
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Mockingbird
          </cite>
          <time
            className="review__date"
            dateTime="2020-07-14T21:00:00.000Z"
          >
            July 15, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8,3
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Super mega exquisitely boring
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Pelagia
          </cite>
          <time
            className="review__date"
            dateTime="2020-05-31T21:00:00.000Z"
          >
            June 1, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        1,5
      </div>
    </div>
  </div>
</div>
`;
